qemu/tests/hotplug_mem.py

```diff
@@ -77,6 +78,7 @@ def start_test(self):
         if not callable(func):
             self.test.error("Unsupported memory operation '%s'" % operation)
         vm = self.env.get_vm(self.params["main_vm"])
+        vm.wait_for_login(timeout=login_timeout)
```

Comment:
If I understand correctly, reboot(or other sub test) happens in sub_test_runner(), so the expected process should be reboot(sub_test_runner()) --> (sleep for a while(with sub_test_wait_time, included in subtest)) --> hotplug(func(vm, target_mem)?) --> check_memory. Is that right? So if you want to hotplug memory and check memory after vm boot up, should be:
````
for target_mem in target_mems.split():
    func(vm, target_mem)
vm.wait_for_login(timeout=login_timeout)
self.check_memory(vm)
```

Reply:
This line is to wait guest to boot up at the beginning, so in the case "during.guest_reboot" the sub test "boot" don't need to wait guest boot up then send reboot cmd, which takes much more time. 
check_memory is demanded to be in the for block by  https://github.com/autotest/tp-qemu/pull/1383.  If add vm.wait_for_login() in the for loop, kind of a waste for the second time check_memory.

Reply:
make sense,  this change is ok for me.

Reply:
Also,  i think,  we need ` wait_for_login` before `self.check_memory` too, to solve the problem you mentioned "check memory in guest would timeout due to guest is reooting."

Reply:
It's ok.  wait_for_login() will be called during processing check_memory(), I believe that's where timeout happens.

Reply:
Since the default timeout value (LOGIN_WAIT_TIMEOUT = 240) in `check_memory` is not configurable, and if we want  keep the current design,  `check_memory` after the change for every target_mem,  How about :
```
need_wait = True
for target_mem in target_mems.split():
    func(vm, target_mem)
    if need_wait:
        vm.wait_for_login(timeout=login_timeout)
        need_wait = False
    self.check_memory(vm)
```